src/read/read_options.rs

```diff
@@ -1,6 +1,6 @@
-use crate::StrictnessLevel;
+use crate::{Context, PDBError, StrictnessLevel};
 
-use super::general::{open_with_options, ReadResult};
+use super::general::ReadResult;
```

Comment:
Is this instead of doing `pub(crate)`? I think I saw a conversation about this. In my opinion, it would be better to simply completely hide the interface, as it means we have more freedom later to change things without affecting the API (which is the may goal of using the options pattern)

Reply:
@douweschulte Thoughts?

I would be in favour of only having `open` and `ReadOptions`, and getting rid of `open_with_options` in the public API, either deleting or marking as `pub(crate)`

Reply:
> @douweschulte Thoughts?
> 
> I would be in favour of only having `open` and `ReadOptions`, and getting rid of `open_with_options` in the public API, either deleting or marking as `pub(crate)`

I think this makes sense, as `open` would be the "magic" function with sane defaults, and `ReadOptions` would be the only API used for fine-grained controls. We may even take a step further and mark `open_*`, `open_*_with_options`, and `open_*_raw` as deprecated and steer users towards `ReadOptions` in these cases.

Reply:
I fully agree. In the end we want to only have the `ReadOptions` and potentially `open` functions. Any function added by this PR can be marked `pub(crate)` any function already present in the previous version has to be marked `deprecated` for at least one major version to help users migrate.

Reply:
I like the sound of that.

Reply:
Should all be deprecated or marked `pub(crate)` now. Please check.